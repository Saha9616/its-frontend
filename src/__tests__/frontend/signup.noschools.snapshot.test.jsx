import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import SignUpView from "@/app/(auth)/signup/page";

// mock server action getSchools that returns nothing
jest.mock("@/actions/getSchools", () => ({
  getSchools: jest.fn().mockResolvedValue(),
}));

it("renders signup page unchanged", async () => {
  const result = await SignUpView();
  const { container } = render(result);
  expect(container).toMatchSnapshot();
});
