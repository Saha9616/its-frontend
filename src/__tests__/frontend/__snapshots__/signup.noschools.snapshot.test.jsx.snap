// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders signup page unchanged 1`] = `
<div>
  <div
    class="flex flex-col space-y-8 justify-center items-center min-h-screen"
  >
    <h1
      class="text-2xl font-semibold"
    >
      ITS Frontend Manager
    </h1>
    <div
      class="rounded-lg border bg-card text-card-foreground shadow-sm min-w-96 md:min-w-[500px]"
    >
      <div
        class="flex flex-col space-y-1.5 p-6"
      >
        <h3
          class="text-2xl font-semibold leading-none tracking-tight text-center"
        >
          Sign Up
        </h3>
      </div>
      <div
        class="p-6 pt-0"
      >
        <form
          class="space-y-4"
        >
          <div
            class="space-y-2"
          >
            <label
              class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
              for=":r0:-form-item"
            >
              Email Address
            </label>
            <div
              aria-describedby=":r0:-form-item-description"
              aria-invalid="false"
              class="flex space-x-2 items-center"
              id=":r0:-form-item"
            >
              <svg
                class="lucide lucide-mail "
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="16"
                  rx="2"
                  width="20"
                  x="2"
                  y="4"
                />
                <path
                  d="m22 7-8.97 5.7a1.94 1.94 0 0 1-2.06 0L2 7"
                />
              </svg>
              <input
                class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                name="email"
                value=""
              />
            </div>
          </div>
          <div
            class="space-y-2"
          >
            <label
              class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
              for=":r1:-form-item"
            >
              Password
            </label>
            <div
              aria-describedby=":r1:-form-item-description"
              aria-invalid="false"
              class="flex space-x-2 items-center"
              id=":r1:-form-item"
            >
              <svg
                class="lucide lucide-lock "
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="11"
                  rx="2"
                  ry="2"
                  width="18"
                  x="3"
                  y="11"
                />
                <path
                  d="M7 11V7a5 5 0 0 1 10 0v4"
                />
              </svg>
              <input
                class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                name="password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="space-y-2"
          >
            <label
              class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
              for=":r2:-form-item"
            >
              Institution/ School
            </label>
            <div
              aria-describedby=":r2:-form-item-description"
              aria-invalid="false"
              class="flex space-x-2 items-center"
              id=":r2:-form-item"
            >
              <svg
                class="lucide lucide-landmark "
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="3"
                  x2="21"
                  y1="22"
                  y2="22"
                />
                <line
                  x1="6"
                  x2="6"
                  y1="18"
                  y2="11"
                />
                <line
                  x1="10"
                  x2="10"
                  y1="18"
                  y2="11"
                />
                <line
                  x1="14"
                  x2="14"
                  y1="18"
                  y2="11"
                />
                <line
                  x1="18"
                  x2="18"
                  y1="18"
                  y2="11"
                />
                <polygon
                  points="12 2 20 7 4 7"
                />
              </svg>
              <button
                aria-autocomplete="none"
                aria-controls="radix-:r3:"
                aria-expanded="false"
                class="flex h-10 w-full items-center justify-between rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1"
                data-placeholder=""
                data-state="closed"
                dir="ltr"
                role="combobox"
                type="button"
              >
                <span
                  name="institution"
                  style="pointer-events: none;"
                  value=""
                >
                  Select Institution
                </span>
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </div>
            <select
              aria-hidden="true"
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
              tabindex="-1"
            />
          </div>
          <div
            class="flex flex-col w-full justify-center items-center text-center"
          >
            <button
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 w-3/4 mt-4"
              type="submit"
            >
              Register
            </button>
          </div>
        </form>
        <div
          class="flex items-center p-6 pt-0 justify-center mt-4"
        >
          <span
            class="text-sm"
          >
            Have an account?
          </span>
          <a
            href="/signin"
          >
            <button
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-10 px-4 py-2 text-sm"
            >
              Login Now
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
